### PR TITLE
Convert AlluxioRuntimeException to IOException

### PR DESCRIPTION
**Pull request by 007DXR**
### What changes are proposed in this pull request?
 if catch `InvalidArgumentRuntimeException`, convert to `IllegalArgumentException` in `AbstractFileSystem.java`.

 if catch other `AlluxioRuntimeException`, convert to `IOException` in `AbstractFileSystem.java`.

### Why are the changes needed?

make users clear about the exception type.

### Does this PR introduce any user facing changes?

none

### CODE REVIEW

**Review by Jackson-Wang-7 — APPROVED** [review]
LGTM, call @jja725 to also review

**Review by jja725 — COMMENTED** [review]
Thanks for the fix! I would recommend taking a look at the AlluxioHdfsException and HDFS interface [doc](https://hadoop.apache.org/docs/stable/api/org/apache/hadoop/fs/FileSystem.html). Then transferring the runtime exception into a more HDFS-specific exception since we only use it in the HDFS interface. MalformedURLException doesn't seem like an HDFS exception.

File: dora/core/common/pom.xml [hunk 1]
```diff
@@ -173,6 +173,30 @@
       <version>${jupiter.version}</version>
       <scope>test</scope>
     </dependency>
+      <dependency>
+          <groupId>org.apache.hadoop</groupId>
```
Comment (jja725) [inline dora/core/common/pom.xml:177]: we would like to put dependency in the root pom and submodules would rely on the root pom change

**Review by jja725 — COMMENTED** [review]
Mostly LGTM, thanks for the fix!

**Review by jja725 — APPROVED** [review]
LGTM, thanks for the fix. @Jackson-Wang-7 do you mind taking another pass on this?

**Comment by jja725** [comment]
@007DXR seems like we miss the `rename` function.

**Comment by Jackson-Wang-7** [comment]
alluxio-bot, merge this please

**Pull request merged**